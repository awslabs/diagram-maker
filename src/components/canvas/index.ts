export { default as Canvas, CanvasProps } from './Canvas';
