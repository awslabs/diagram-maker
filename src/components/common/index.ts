export { default as ComposeView } from './ComposeView';
