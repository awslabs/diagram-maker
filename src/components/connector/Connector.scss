$handleSize: 32px;
$dotSize: 14px;

.dm-connector {
  position: absolute;
  top:0;
  left:0;
  width: $handleSize;
  height: $handleSize;
  border-radius: 100%;
  margin-top: -$handleSize/2;
  margin-left: -$handleSize/2;

  &.dm-connector-output {
    cursor: pointer;
  }

  .dm-handle {
    box-sizing: border-box;
    background: #16191f;
    transition: background-color .4s ease, border .4s ease, transform .25s ease;
    width: $dotSize;
    height: $dotSize;
    margin: ($handleSize - $dotSize) / 2;
    border-radius: 100%;
  }
}
