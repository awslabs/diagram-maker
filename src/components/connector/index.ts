export {
  default as Connector, ConnectorProps, ConnectorType
} from './Connector';
