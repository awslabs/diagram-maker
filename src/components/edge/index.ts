export { EdgeStyle } from './EdgeCurve';
export { default as Edge, EdgeProps } from './Edge';
export { default as EdgeBadge, EdgeBadgeProps } from './EdgeBadge';
export { default as PotentialEdge, PotentialEdgeProps } from './PotentialEdge';
