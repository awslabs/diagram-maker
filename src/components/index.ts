export { Canvas, CanvasProps } from './Canvas';
export { default as ComposeView } from './common/ComposeView';
export {
  default as Connector,
  ConnectorProps,
  ConnectorType
} from './connector/Connector';
export {
  default as ContextMenu,
  ContextMenuProps
} from './contextMenu/ContextMenu';
export { Node, NodeProps, PotentialNode, PotentialNodeProps } from './node';
export { Panel, PanelProps } from './panel';
export { SelectionMarquee, SelectionMarqueeProps } from './selectionMarquee';
export { View, ViewProps } from './view/View';
export { getConnectedView, ConnectedViewOwnProps } from './view/ConnectedView';
export { Workspace, WorkspaceProps } from './Workspace';

export * from './edge/EdgeCurve';
export { default as Edge, EdgeProps } from './edge/Edge';
export { default as EdgeBadge, EdgeBadgeProps } from './edge/EdgeBadge';
export {
  default as PotentialEdge,
  PotentialEdgeProps
} from './edge/PotentialEdge';

export {
  default as GridPattern,
  GridPatternProps
} from './canvas/patterns/GridPattern';

export {
  default as DotPattern,
  DotPatternProps
} from './canvas/patterns/DotPattern';

export { render, destroy } from './renderUtils';
