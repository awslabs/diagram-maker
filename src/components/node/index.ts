export { default as Node, NodeProps } from './Node';
export { default as PotentialNode, PotentialNodeProps } from './PotentialNode';
