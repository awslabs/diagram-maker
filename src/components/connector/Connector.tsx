import * as classnames from 'classnames';
import * as Preact from 'preact';

import { DiagramMakerComponentsType } from 'diagramMaker/service/ui/types';
import { Position } from 'diagramMaker/state/types';

import './Connector.scss';

export enum ConnectorType {
  INPUT,
  OUTPUT
}

export interface ConnectorProps {
  id: string;
  position: Position;
  type: ConnectorType;
}

export default class Connector extends Preact.Component<ConnectorProps, {}> {
  public render(): JSX.Element {
    const { id, position, type } = this.props;
    const { x, y } = position;
    const classes = classnames('dm-connector', {
      'dm-connector-input': type === ConnectorType.INPUT,
      'dm-connector-output': type === ConnectorType.OUTPUT
    });
    const transform = `translate3d(${x}px, ${y}px, 0)`;
    const interactionProps: { [index: string]: string | boolean } = {
      'data-id': id, 'data-type': DiagramMakerComponentsType.NODE_CONNECTOR
    };
    if (type === ConnectorType.INPUT) {
      interactionProps['data-dropzone'] = true;
    } else {
      interactionProps['data-draggable'] = true;
    }

    return (
      <div
        className={classes}
        style={{ transform }}
        {...interactionProps}
      >
        <div className="dm-handle" />
      </div>
    );
  }
}
