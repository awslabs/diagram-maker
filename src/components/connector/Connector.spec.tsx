import * as Preact from 'preact';
import { shallow } from 'preact-render-spy';

import { Connector, ConnectorType } from 'diagramMaker/components/connector';

describe('Connector', () => {
  it('renders at the given position with output data attributes', () => {
    const connector = shallow(
      <Connector
        id="test"
        position={{ x:150, y:300 }}
        type={ConnectorType.OUTPUT}
      />
    );

    expect(connector).toMatchSnapshot();
  });

  it('renders at the given position with input data attributes', () => {
    const connector = shallow(
      <Connector
        id="test"
        position={{ x:150, y:300 }}
        type={ConnectorType.INPUT}
      />
    );

    expect(connector).toMatchSnapshot();
  });
});
