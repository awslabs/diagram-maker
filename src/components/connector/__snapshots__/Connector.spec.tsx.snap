// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connector renders at the given position with input data attributes 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-connector dm-connector-input"
  style="transform: translate3d(150px, 300px, 0);"
  data-id="test"
  data-type="DiagramMaker.Connector"
  data-dropzone={true}
>
  <div class="dm-handle"></div>
</div>

`;

exports[`Connector renders at the given position with output data attributes 1`] = `
preact-render-spy (1 nodes)
-------
<div
  class="dm-connector dm-connector-output"
  style="transform: translate3d(150px, 300px, 0);"
  data-id="test"
  data-type="DiagramMaker.Connector"
  data-draggable={true}
>
  <div class="dm-handle"></div>
</div>

`;
